### ISSUE TITLE
Chunked Transfer broken

### ISSUE THREAD

**Issue by fkollmann** [zevenet/zproxy#5]
Hi, with the current v0.2 version, the chunked transfer seems to be broken:

```sh
curl --output /dev/null -H 'Transfer-Encoding: chunked' -H 'Docker-Distribution-API-Version: registry/2.0' https://docker.devops.usu.group/v2/
```

The transfer stops a 0 bytes. This results in docker pulls and pushes to break (sometimes they work, sometimes not).

Best Regards, Felix

**Comment by abdessamad-zevenet** [comment]
Hi Felix,

Thank you for reporting the issue, It might be the issue I'm working on. I've just pushed a new branch with some IO handling changes, could test on that please?

Kind regards!

**Comment by fkollmann** [comment]
Hi, I just tried the new version. The issue persists and it has some side effects, to I rolled back to the v0.2.

One one case of my testing, I got this response:

```sh
curl -H 'Transfer-Encoding: chunked' -H 'Docker-Distribution-API-Version: registry/2.0' https://docker.devops.usu.group/v2/
curl: (52) Empty reply from server
```

And for regular requests, I got a new side effect:

```sh
docker login docker.devops.usu.group -u xxxxxx
Password: 
Error response from daemon: Get https://docker.devops.usu.group/v2/: read tcp x.x.x.x:34684->62.96.81.133:443: read: connection reset by peer
```

Best Regards, Felix

**Comment by fkollmann** [comment]
I just saw that you repushed the last change. Let me try again, then.

**Comment by fkollmann** [comment]
Okay, I have to go back on my previous statements. It look my test case here is not 100% valid. With the provided branch the docker pulls seem to work fine. I will do some additional testing... Best Regards, Felix

**Comment by fkollmann** [comment]
Pushing seems to work fine, too. Thanks man!! This really helped :) . Best Regards, Felix

**Comment by abdessamad-zevenet** [comment]
Yes, I had to re-push due to some testing code I had to remove in order to work as expected, sorry!

Thank you for testing, this will be merged soon to the master branch, please let me know if you have any issue!

Kind regards!

**Comment by abdessamad-zevenet** [comment]
I 've just pushed the latest changes to the master branch, please if you could update and test if everything works for you.

**Comment by fkollmann** [comment]
v0.2.1, right? I will try this evening.

**Comment by fkollmann** [comment]
I used the official version from the deb repository and it seems to work fine, so far. Our first tests were successful. 👍

**Comment by abdessamad-zevenet** [comment]
Great, thank your for the confirmation!
